lldb/source/DataFormatters/TypeSummary.cpp

```diff
@@ -145,15 +148,34 @@ std::string CXXFunctionSummaryFormat::GetDescription() {
   return std::string(sstr.GetString());
 }
 
+std::string CXXFunctionSummaryFormat::GetName() {
+  return m_description;
+}
+
+std::string CXXFunctionSummaryFormat::GetSummaryKindName() {
+  return "c++";
+}
+
 ScriptSummaryFormat::ScriptSummaryFormat(const TypeSummaryImpl::Flags &flags,
                                          const char *function_name,
                                          const char *python_script)
     : TypeSummaryImpl(Kind::eScript, flags), m_function_name(),
       m_python_script(), m_script_function_sp() {
-  if (function_name)
+  std::string sstring;
```

Comment:
Why not operate on `m_script_formatter_name ` directly?

Reply:
No strong reason, my only concern was emplacing to `m_script_formatter_name` was that I needed to trim the leading zeros, but writing this out there really is no reason.